core/dictgen/src/BaseSelectionRule.cxx

```diff
@@ -218,7 +218,7 @@ BaseSelectionRule::EMatchType BaseSelectionRule::Match(const clang::NamedDecl *d
                                   ROOT::TMetaUtils::GetUnderlyingRecordDecl(typedefNameDecl->getUnderlyingType());
       }
 
-   if (! isTypedefNametoRecordDecl && fCXXRecordDecl !=0 && fCXXRecordDecl != (void*)-1) {
+   if (! isTypedefNametoRecordDecl && !fCXXRecordDecl && !fCXXRecordDeclLookupFailed) {
```

Comment:
I think here is `if (! isTypedefNametoRecordDecl && fCXXRecordDecl !=0 !fCXXRecordDeclLookupFailed) {`